src/util.js

```diff
@@ -276,6 +276,51 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
+// for debug methods
+function hookTo(prototypeObject, methodName, extension) {
+    var existingMethodImpl = prototypeObject[methodName];
+    if (typeof existingMethodImpl === "function" &&
+        existingMethodImpl.extensions) {
+        var extensions = existingMethodImpl.extensions;
+        extensions[extension.toString()] = extension;
+        // Performance optimization: remove noops
+        for (var key in extensions) {
+            if (extensions.hasOwnProperty(key) &&
+                extensions[key].toString().replace(/\s/g, "") ===
+                "function(){}") {
+                delete extensions[key];
+            }
+        }
+    } else if (typeof existingMethodImpl === "undefined") {
+        prototypeObject[methodName] = function () {
+            var extensions = prototypeObject[methodName].extensions;
+            for (var key in extensions) {
+                if (extensions.hasOwnProperty(key)) {
+                    extensions[key].apply(this, arguments);
+                }
+            }
+        };
+        // Using object instead of array of functions to allow optimizations
```

Comment:
As far as I understand saving functions on object property and not in array allows JS execution engine to optimize out useless frame in call stack (or another part of call stack). If it would be saved in array longStackTraces test would fail with message "Expected 2 jumps but saw 3 jumps".